spec/index.bs

```diff
@@ -155,22 +159,40 @@ See [[#errors-and-debugging]] for more information about error handling.
 WebGPU [=shader=]s are executed by the compute units inside GPU hardware. In native APIs,
 some of the shader instructions may result in undefined behavior on the GPU.
 In order to address that, the shader instruction set and its defined behaviors are
-strictly defined by WebGPU. When a shader is provided, the WebGPU implementation has to validate it
+strictly defined by WebGPU. When a shader is provided to {{GPUDevice/createShaderModule()}},
+the WebGPU implementation has to validate it
 before doing any translation (to platform-specific shaders) or transformation passes.
 
+## Uninitalized data ## {#security-uninitialized}
+
+Generally, allocating new memory may expose the leftover data of other applications running on the system.
+In order to address that, WebGPU initializes all the resources to zero.
```

Comment:
Should we mention something about the clearing being expected to be done lazily?

Reply:
Should be better now. Please re-open if more changes are needed.